File: laba5/c++/Line1.cpp
#include "Line1.h"
#include <iostream>

using namespace std;

float Line1::getX() // метод отримання даних об'єкта
{
	return x;
}

float Line1::getX1() // метод отримання даних об'єкта
{
	return x1;
}

float Line1::getY() // метод отримання даних об'єкта
{
	return y;
}

float Line1::getY1() // метод отримання даних об'єкта
{
	return y1;
}

Line1::Line1(float a, float b, float c, float d) : Line(a, b, c, d) // конструктор з параметрами
{

}

double Line1::Len() // метод обчислення довжини
{
	return 2 * Line::Len();
}

void Line1::Draw() //виведення координат на екран
{
	cout << "start point = " << "(" << x << "," << y << ")" << endl;
	cout << "end point = " << "(" << x1 << "," << y1 << ")" << endl;

}

File: laba5/c#/Line1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OOP5
{
    class Line1 : Line
    {

        public float getX(){ return x; }//метод отримання даних об'єкта
        public float getY(){ return y; }//метод отримання даних об'єкта
        public float getX1(){ return x1; }//метод отримання даних об'єкта
        public float getY1(){ return y1; }//метод отримання даних об'єкта

        public Line1(float x, float y, float x1, float y1) ://конструктор з параметрами
            base(x, y, x1, y1)
        {
          
        }

        public override double Len() //методи збільшення довжини вдвічі
        {
                return 2 * base.Len();
        }

        public void Draw() //виведення координат на екран
        {
            Console.WriteLine("Line start:({0},{1}) Line end:({2},{3})", x, y, x1, y1);
        }

    }
}
